code/modules/mob/living/simple_animal/hostile/cockroach.dm

```diff
@@ -63,25 +63,54 @@
 	. = ..()
-	if(ismob(AM))
-		if(isliving(AM))
-			var/mob/living/A = AM
-			if(A.mob_size > MOB_SIZE_SMALL && !(A.movement_type & FLYING))
-				if(HAS_TRAIT(A, TRAIT_PACIFISM))
-					A.visible_message("<span class='notice'>[A] carefully steps over [src].</span>", "<span class='notice'>You carefully step over [src] to avoid hurting it.</span>")
-					return
-				if(prob(squish_chance))
-					A.visible_message("<span class='notice'>[A] squashed [src].</span>", "<span class='notice'>You squashed [src].</span>")
-					adjustBruteLoss(1) //kills a normal cockroach
-				else
-					visible_message("<span class='notice'>[src] avoids getting crushed.</span>")
-	else
-		if(isstructure(AM))
+	if(isliving(AM))
+		var/mob/living/A = AM
+		if(A.mob_size > MOB_SIZE_SMALL && !(A.movement_type & FLYING))
+			if(HAS_TRAIT(A, TRAIT_PACIFISM))
+				A.visible_message("<span class='notice'>[A] carefully steps over [src].</span>", "<span class='notice'>You carefully step over [src] to avoid hurting it.</span>")
+				return
 			if(prob(squish_chance))
-				AM.visible_message("<span class='notice'>[src] is crushed under [AM].</span>")
-				adjustBruteLoss(1)
+				A.visible_message("<span class='notice'>[A] squashed [src].</span>", "<span class='notice'>You squashed [src].</span>")
+				adjustBruteLoss(1) //kills a normal cockroach
 			else
 				visible_message("<span class='notice'>[src] avoids getting crushed.</span>")
+	else if(isstructure(AM))
+		if(prob(squish_chance))
+			AM.visible_message("<span class='notice'>[src] is crushed under [AM].</span>")
+			adjustBruteLoss(1)
+		else
+			visible_message("<span class='notice'>[src] avoids getting crushed.</span>")
 
 /mob/living/simple_animal/hostile/cockroach/ex_act() //Explosions are a terrible way to handle a cockroach.
 	return
+
+/mob/living/simple_animal/hostile/cockroach/hauberoach
+	name = "hauberoach"
+	desc = "Is that cockroach wearing a tiny yet immaculate replica 19th century Prussian spiked helmet? ...Is that a bad thing?"
+	icon_state = "hauberoach"
+	attack_verb_continuous = "rams its spike into"
+	attack_verb_simple = "ram your spike into"
+	melee_damage_lower = 5
+	melee_damage_upper = 20
+	obj_damage = 20
+	gold_core_spawnable = HOSTILE_SPAWN
+	attack_sound = 'sound/weapons/bladeslice.ogg'
+	faction = list("hostile")
+	sharpness = IS_SHARP
+	squish_chance = 0 // manual squish if relevant
+
+/mob/living/simple_animal/hostile/cockroach/hauberoach/ComponentInitialize()
+	. = ..()
+	AddComponent(/datum/component/caltrop, 10, 15, 100, (CALTROP_BYPASS_SHOES | CALTROP_SILENT))
+
+/mob/living/simple_animal/hostile/cockroach/hauberoach/Crossed(var/atom/movable/AM)
+	var/mob/living/A = AM
+	if(istype(A) && A.mob_size > MOB_SIZE_SMALL && !(A.movement_type & FLYING))
+		if(!HAS_TRAIT(A, TRAIT_PIERCEIMMUNE))
+			A.visible_message("<span class='danger'>[A] steps onto [src]'s spike!</span>", "<span class='userdanger'>You step onto [src]'s spike!</span>")
+		else if(HAS_TRAIT(A, TRAIT_PACIFISM))
```

Comment:
imo, you should swap the position of this with the position of the pierceimmune check, if possible

Reply:
since when did the prussians honor peace treaties?

Reply:
it's less that and more that if a pacifist is intentionally trying to NOT step on the hauberoach, then it shouldn't matter if they're pierceimmune or not.

Reply:
you could add a special caltrop flag that makes it not affect pacifists.

Reply:
No